src/main/java/org/opensearch/knn/index/codec/nativeindex/remote/RemoteIndexBuildStrategy.java

```diff
@@ -126,19 +140,29 @@ public void buildAndWriteIndex(BuildIndexParams indexInfo) throws IOException {
             log.debug("Repository write took {} ms for vector field [{}]", time_in_millis, indexInfo.getFieldName());
 
             RemoteIndexClient client = RemoteIndexClientFactory.getRemoteIndexClient();
-            RemoteBuildRequest request = new RemoteBuildRequest(indexSettings, indexInfo, getRepository().getMetadata(), blobName);
+            RemoteBuildRequest buildRequest = new RemoteBuildRequest(
+                indexSettings,
+                indexInfo,
+                repository.getMetadata(),
+                blobPath.add(blobName).buildAsString(),
```

Comment:
```suggestion
                blobPath.buildAsString() + blobName,
```
1. `BlobPath` class is only intended to represent directories, not the file itself (hence the ending `/`), so let's not mis-use it in this way.
2. You don't need the logic to get rid of the ending `/` this way

Reply:
Nice! Thanks, fixed.